### ISSUE TITLE
Compatibility Issue with mongoose version 6.0.x

### ISSUE THREAD

**Issue by Kaizennnnn** [Automattic/mongoose#10805]
when I upgraded my mongoose version to 6.0.7, I got some error like below

![image](https://user-images.githubusercontent.com/91593851/135226490-d23eebde-444a-4116-a582-8685aa215954.png)
the error here is:

![image](https://user-images.githubusercontent.com/91593851/135226881-6c07c51b-5aed-4f33-8243-689eb194ff26.png)

I'm using 
![image](https://user-images.githubusercontent.com/91593851/135227472-4485c0b8-728d-4cde-b414-5fbc8be379d4.png)

Can someone help me solve this problem? Thanks

**Comment by IslandRhythms** [comment]
```javascript
const mongoose = require('mongoose');



async function run() {
    await mongoose.connect('mongodb://localhost:27017/', {useNewUrlParser: true,
    useUnifiedTopology: true,});
    await mongoose.connection.dropDatabase();
   const session = await mongoose.connection.startSession({defaultTransactionOptions: {
        writeConcern: {w: 1},
        readConcern: {level: 'snapshot'}
    }});
    session.startTransaction();
    mongoose.sessions.set(id, session);
}

run();
```

**Comment by Kaizennnnn** [comment]
Hi @IslandRhythms I tried your way but still not solved. the main point here is where readConcern: {level: 'snapshot'} it gives an error: 'property 'toJSON' is missing in type '{ level: string; }' in readConcern mongoDb'
Is your mongoose version is 6.x?
I appreciate all your comments

**Comment by IslandRhythms** [comment]
I'm not getting that error so it must be something with your set up. I am on mongoose latest

**Comment by Uzlopak** [comment]
It was merged in mongodb 4.2, so we can close this issue :)
